examples/docker/stable_diffusion_2_0_docker.py

```diff
@@ -0,0 +1,273 @@
+This file demonstrates this by creating a basic Stable Diffusion Pipeline with a custom
+environment and deploying the docker image.
+
+Run this code in this directory:
+
+env HF_TOKEN="your hugging face token" python stable_diffusion_2_0_docker.py
+
+and then start the docker containers created:
+
+sudo docker compose up -d
+
+you can test the running images:
+
+curl -X POST 'http://localhost:5010/v2/runs' \
+-H 'Content-Type: application/json' \
+-d '{
+    "pipeline_id": "sd-2_0",
+    "data": [
+        [
+            {
+                "text_in": "Mountain winds, and babbling springs, and moonlight seas"
+            }
+        ],
+        {
+            "num_samples": 1,
+            "num_inference_steps": 25,
+            "width": 768,
+            "height": 768
+        }
+    ]
+}'
+"""
+
+import os
+import random
+from typing import Optional, TypedDict
+
+import numpy as np
+import torch
+from cloudpickle import dumps
+from diffusers import DiffusionPipeline, DPMSolverMultistepScheduler
+from diffusers.utils import logging
+from dill import loads
+
+from pipeline import (
+    Pipeline,
+    PipelineFile,
+    Variable,
+    docker,
+    pipeline_function,
+    pipeline_model,
+)
+from pipeline.objects.environment import Environment
+
+# the below two functions don't seem to do their job
+logging.disable_progress_bar()
+logging.set_verbosity_error()
+
+
+scheduler = DPMSolverMultistepScheduler.from_pretrained(
+    "stabilityai/stable-diffusion-2", subfolder="scheduler"
+)
+model = DiffusionPipeline.from_pretrained(
+    "stabilityai/stable-diffusion-2",
+    use_auth_token=os.environ["HF_TOKEN"],
+    scheduler=scheduler,
+    revision="fp16",
+    torch_dtype=torch.float16,
+    safety_checker=None,
+)
+
+temp_path = "./temporary.model"
+with open(temp_path, "wb") as tmp_file:
+    tmp_file.write(dumps(model))
+
+#
+# pipeline.ai logic
+#
+
+
+def seed_everything(seed: int) -> int:
+    os.environ["PL_GLOBAL_SEED"] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    return seed
+
+
+def get_aspect_ratio(width: int, height: int) -> float:
+    return float(width / height)
+
+
+class PromptShape(TypedDict):
+    text_in: str
+    seed: Optional[int]
+
+
+class BatchKwargsShape(TypedDict):
+    num_samples: Optional[int]
+    height: Optional[int]
+    width: Optional[int]
+    seed: Optional[int]
+    num_inference_steps: Optional[int]
+    guidance_scale: Optional[float]
+    eta: Optional[float]
+    randomise_seed: Optional[bool]
+
+
+@pipeline_model
+class StableDiffusionTxt2ImgModel:
+    @pipeline_function
+    def predict(
+        self, prompts: list[PromptShape], batch_kwargs: BatchKwargsShape
+    ) -> list[list[str]]:
+        print("🦁 run start")
+        print(prompts)
+        import base64
+        import random
+        from io import BytesIO
+
+        default_batch_kwargs = {
```

Comment:
Because the inputs are Pydantic models, could you define the defaults in the model defintions instead?

Reply:
I'm going to update this completely in a follow up PR, not happy with the system at all.